syntax = "proto3";

import "market.proto";

service BuyerService {
    rpc SearchItem(SearchItemRequest) returns (SearchItemResponse) {}

    rpc BuyItem(BuyItemRequest) returns (BuyItemResponse) {}

    rpc AddToWishList(AddToWishListRequest) returns (AddToWishListResponse) {}

    rpc RateItem(RateItemRequest) returns (RateItemResponse) {}
}


message SearchItemRequest {
    string item_name = 1;
    enum Category {
        ELECTRONICS = 0;
        FASHION = 1;
        OTHERS = 2;
        ANY = 3;
    }
    string buyer_address = 2;
}


message SearchItemResponse {
    message Item {
        int32 item_id = 1;
        string product_name = 2;
        enum Category {
            ELECTRONICS = 0;
            FASHION = 1;
            OTHERS = 2;
        }
        int32 price = 3;
        int32 quantity_remaining = 4;
        float rating = 5;
        string seller_address = 6;
    }
    repeated Item items = 1;
}


message BuyItemRequest {
    int32 item_id = 1;
    int32 quantity = 2;
    string buyer_address = 3;
}

message BuyItemResponse {
    enum Status {
        SUCCESS = 0;
        FAILED = 1;
    }
    Status status = 1;
}

message AddToWishListRequest {
    int32 item_id = 1;
    string buyer_address = 2;
}

message AddToWishListResponse {
    enum Status {
        SUCCESS = 0;
        FAILED = 1;
    }
    Status status = 1;
}

message RateItemRequest {
    int32 item_id = 1;
    string buyer_address = 2;
    int32 rating = 3;
}

message RateItemResponse {
    enum Status {
        SUCCESS = 0;
        FAILED = 1;
    }
    Status status = 1;
}
